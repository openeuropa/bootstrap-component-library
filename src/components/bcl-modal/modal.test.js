import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";

import demoData from "@openeuropa/bcl-data-modal/data";

const template = "@oe-bcl/bcl-modal/modal.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - modal", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test("renders correctly small", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, size: "sm" })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with fullscreen breakpoint", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, fullscreen_responsive: "sm" })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with static backdrop", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, static_backdrop: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with verticaly centered modal", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, verticaly_centered: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with scrollable modal", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, scrollable: true })
    ).resolves.toMatchSnapshot();
  });
});
