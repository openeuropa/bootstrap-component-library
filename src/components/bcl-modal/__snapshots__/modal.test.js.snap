// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - modal renders correctly 1`] = `
<jest>
  <div
    aria-hidden="true"
    aria-label="exampleModalLabel"
    class="modal fade"
    id="exampleModal"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
            id="exampleModalLabel"
          >
            Modal title
          </h5>
          <button
            aria-label="Close"
            class="btn-close"
            data-bs-dismiss="modal"
            type="button"
          />
        </div>
        <div
          class="modal-body"
        >
          Modal body
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-secondary"
            data-bs-dismiss="modal"
            type="button"
          >
            Close
          </button>
          <button
            class="btn btn-primary"
            type="button"
          >
            Save changes
          </button>
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - modal renders correctly with scrollable modal 1`] = `
<jest>
  <div
    aria-hidden="true"
    aria-label="exampleModalLabel"
    class="modal fade"
    id="exampleModal"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-dialog-scrollable"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
            id="exampleModalLabel"
          >
            Modal title
          </h5>
          <button
            aria-label="Close"
            class="btn-close"
            data-bs-dismiss="modal"
            type="button"
          />
        </div>
        <div
          class="modal-body"
        >
          Modal body
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-secondary"
            data-bs-dismiss="modal"
            type="button"
          >
            Close
          </button>
          <button
            class="btn btn-primary"
            type="button"
          >
            Save changes
          </button>
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - modal renders correctly with static backdrop 1`] = `
<jest>
  <div
    aria-hidden="true"
    aria-label="exampleModalLabel"
    class="modal fade"
    id="exampleModal"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
      data-bs-backdrop="static"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
            id="exampleModalLabel"
          >
            Modal title
          </h5>
          <button
            aria-label="Close"
            class="btn-close"
            data-bs-dismiss="modal"
            type="button"
          />
        </div>
        <div
          class="modal-body"
        >
          Modal body
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-secondary"
            data-bs-dismiss="modal"
            type="button"
          >
            Close
          </button>
          <button
            class="btn btn-primary"
            type="button"
          >
            Save changes
          </button>
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - modal renders correctly with verticaly centered modal 1`] = `
<jest>
  <div
    aria-hidden="true"
    aria-label="exampleModalLabel"
    class="modal fade"
    id="exampleModal"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-dialog-centered"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
            id="exampleModalLabel"
          >
            Modal title
          </h5>
          <button
            aria-label="Close"
            class="btn-close"
            data-bs-dismiss="modal"
            type="button"
          />
        </div>
        <div
          class="modal-body"
        >
          Modal body
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-secondary"
            data-bs-dismiss="modal"
            type="button"
          >
            Close
          </button>
          <button
            class="btn btn-primary"
            type="button"
          >
            Save changes
          </button>
        </div>
      </div>
    </div>
  </div>
</jest>
`;
